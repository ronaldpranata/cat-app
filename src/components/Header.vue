<template>
    <div id="nav">
        <span v-for="(item,index) in links" :key="index">
            <router-link  :to="item.url">{{ t(item.name) }} </router-link> <span v-if="index!=(links.length-1)">|</span>
        </span>
    </div>
</template>

<script>
import { defineComponent } from 'vue';
import { useI18n } from 'vue-i18n';

export default defineComponent({
  name: 'Header',
  props: {
    links: Array,
  },
  setup() {
    const { t } = useI18n();
    return { t };
  },
});
</script>
<style scoped lang="scss">
#nav {
  padding: 30px;

  a {
    font-weight: bold;
    color: #2c3e50;

    &.router-link-exact-active {
      color: #42b983;
    }
  }
}
</style>
