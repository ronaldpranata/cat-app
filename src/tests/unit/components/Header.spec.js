import { shallowMount } from '@vue/test-utils';
import { createI18n } from 'vue-i18n';
import Header from '@/components/Header.vue';

const i18n = createI18n({
  // vue-i18n options here ...
});
const factory = () => shallowMount(Header, {
  propsData: {
    links:
      [
        {
          url: '/',
          name: 'home',
        },
        {
          url: '/about',
          name: 'about',
        },
      ],
  },
  global: {
    plugins: [i18n],
  },
});

describe('Header', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
