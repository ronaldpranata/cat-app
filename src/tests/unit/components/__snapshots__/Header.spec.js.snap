// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<div class="header">
  <div id="nav"><span><router-link to="/">home</router-link> <span>|</span></span><span><router-link to="/about">about</router-link> <!--v-if--></span></div><img alt="Vue logo" src="">
</div>
`;
